mods/tech/glassworking.lua

```diff
@@ -366,3 +366,131 @@ crafting.register_recipe({
 	output = "tech:clear_glass_mix 8",
 	items = {'tech:potash 1', 'tech:quicklime 1', 'nodes_nature:sand 6'},
 	level = 1,
+always_known = true,
+})
+
+-- Pane casting tray - heat up a glass ingot above it to cast a pane
+minetest.register_node("tech:pane_tray",
+{
+	description = "Pane Casting Tray",
+	tiles = {"tech_iron.png"},
+	drawtype = "nodebox",
+	node_box = 
+	{
+		type = "fixed",
+		fixed = {
+			{-0.5, -0.5, -0.5, 0.5, -0.4, 0.5},
+			{-0.5, -0.4, -0.5, -0.4, -0.3, 0.5},
+			{0.5, -0.4, -0.5, 0.4, -0.3, 0.5},
+			{-0.5, -0.4, -0.5, 0.5, -0.3, -0.4},
+			{-0.5, -0.4, 0.5, 0.5, -0.3, 0.4}
+		}
+		
+	},
+	stack_max = minimal.stack_max_bulky * 2,
+	paramtype2 = "facedir",
+	groups = {cracky = 3},
+	sunlight_propagates = true,
+})
+
+-- Trays with glass panes
+minetest.register_node("tech:pane_tray_green",
+{
+	description = "Pane Casting Tray With Green Glass Pane",
+	tiles = {"tech_tray_green.png", "tech_iron.png", "tech_iron.png", "tech_iron.png", "tech_iron.png", "tech_iron.png"},
+	drawtype = "nodebox",
+	node_box = 
+	{
+		type = "fixed",
+		fixed = {
+			{-0.5, -0.5, -0.5, 0.5, -0.3, 0.5},
+			{-0.5, -0.4, -0.5, -0.4, -0.3, 0.5},
+			{0.5, -0.4, -0.5, 0.4, -0.3, 0.5},
+			{-0.5, -0.4, -0.5, 0.5, -0.3, -0.4},
+			{-0.5, -0.4, 0.5, 0.5, -0.3, 0.4}
+		}
+		
+	},
+	stack_max = minimal.stack_max_bulky * 2,
+	paramtype2 = "facedir",
+	groups = {cracky = 3},
+	sunlight_propagates = true,
+	drop = {
+		max_items = 2,
+		items = {
+			{items = {"tech:pane_tray"}},
+			{items = {"tech:pane_green"}},
+		}
+	}
+
+})
+
+minetest.register_node("tech:pane_tray_clear",
+{
+	description = "Pane Casting Tray With Clear Glass Pane",
+	tiles = {"tech_tray_clear.png", "tech_iron.png", "tech_iron.png", "tech_iron.png", "tech_iron.png", "tech_iron.png"},
+	drawtype = "nodebox",
+	node_box = 
+	{
+		type = "fixed",
+		fixed = {
+			{-0.5, -0.5, -0.5, 0.5, -0.3, 0.5},
+			{-0.5, -0.4, -0.5, -0.4, -0.3, 0.5},
+			{0.5, -0.4, -0.5, 0.4, -0.3, 0.5},
+			{-0.5, -0.4, -0.5, 0.5, -0.3, -0.4},
+			{-0.5, -0.4, 0.5, 0.5, -0.3, 0.4}
+		}
+		
+	},
+	stack_max = minimal.stack_max_bulky * 2,
+	paramtype2 = "facedir",
+	groups = {cracky = 3},
+	sunlight_propagates = true,
+	drop = {
+		max_items = 2,
+		items = {
+			{items = {"tech:pane_tray"}},
+			{items = {"tech:pane_clear"}},
+		}
+	}
+
+})
+
+-- Pane Casting ABM
```

Comment:
https://github.com/jeremyshannon/Exile/pull/16/commits/8e9ef038e1a62cd52f82a03b0bf0fcc12ccddbb6#diff-1349893b2c4e619fa3e24ea3b7d33b1f1a46b073bc54ae913bf55e287de1eda8R459
>-- Pane Casting ABM

Why an ABM? My first thought is that it seems like a node timer would be more appropriate.

Reply:
Yes, that's a dumb mistake on my part too. At first I thought it will be more efficient, but seeing as pane casting trays probably won't be too common I guess a node timer is better.

Reply:
Yeah, it's probably not a huge difference, but as I understand it node timers are usually better for performance unless you're using a ton of them, while ABMs are best used for nodes that are going to exist in large numbers like a biome full of flowers that all need to get updated whenever players move around.
 This is fine for now, though.